docs/source/developer_guide/core/transport.rst

```diff
@@ -3,369 +3,59 @@ Transport plugins
 
-Generic transport class
------------------------
-
-.. automodule:: aiida.transports
-   :members:
-   :noindex:
-   :special-members: __enter__, __exit__,__unicode__
-
-Existing plugins
-----------------
-.. automodule:: aiida.transports.plugins.ssh
-   :members:
-   :noindex:
-   :special-members: __enter__, __exit__,__unicode__
-
-
-.. automodule:: aiida.transports.plugins.local
-   :members:
-   :noindex:
-   :special-members: __enter__, __exit__,__unicode__
 
 Developing a plugin
 -------------------
 
 The transport class is actually almost never used directly by the user.
-It is mostly utilized by the ExecutionManager, that use the transport plugin to connect to the remote computer to manage the calculation.
-The ExecutionManager has to be able to use always the same function, or the same interface, regardless of which kind of connection is actually really using. 
+It is mostly utilized by the engine, that uses the transport plugin to connect to the remote computer where the calculation, that it is managing, is running.
+The engine has to be able to use always the same methods regardless of which kind of transport is required to connect to the computer in question.
 
 The generic transport class contains a set of minimal methods that an implementation must support, in order to be fully compatible with the other plugins.
-If not, a NotImplementedError will be raised, interrupting the managing of the calculation or whatever is using the transport plugin.
-
-Since it is important that all plugins have the same interface, or the same response behavior, a set of generic tests has been written (alongside with set of tests that are implementation specific).
-After **every** modification, or when implementing a new plugin, it is crucial to run the tests and verify that everything is passed.
-The modification of tests possibly means breaking back-compatibility and/or modifications to every piece of code using a transport plugin.
+If not, a ``NotImplementedError`` will be raised, interrupting the managing of the calculation or whatever is using the transport plugin.
 
-If an unexpected behavior is observed during the usage, the way of fixing it is:
+Since it is important that all plugins have the same interface, or the same response behavior, a set of generic tests has been written (alongside with a set of tests that are implementation specific).
+After **every** modification, or when implementing a new plugin, it is crucial to run the tests and verify that all of them pass.
+The modification of tests possibly means breaking backwards compatibility and/or modifications to every piece of code using a transport plugin.
 
-1) Write a new test that shows the problem (one test for one problem when possible)
+If unexpected behavior is observed during the usage, the way of fixing it is:
```

Comment:
Also this comment on how to write tests can be removed IMO, since this is true for everything and not only for Transport, and here it distracts the reader

Reply:
Agree, deleted